Automattic/newspack-plugin
# feat(ia): Newspack > Settings > Syndication Wizard

### All Submissions:


### Changes proposed in this Pull Request:

* Migrated Syndication components to new Information Architecture.
   * Converted to TypeScript.
   * Refactored to use new components and hooks
* Added new hook `useWizardApiFetchToggle` for toggle (on/off) specific cards. See [`<Rss />`](https://github.com/Automattic/newspack-plugin/blob/feat/ia-settings-syndication/assets/wizards/newspack/views/settings/syndication/rss.tsx) component for usage.
* Refactored types to be more reusable i.e. `WizardTab`, `ApiMethods` and `PluginResponse`.
* Added `Configured` property to `plugins/configure` endpoint.
* Return `true` (for `is_configured` method) if no associated configuration manager class is found.

**Plugin Action Cards:**
* Refactored schema for `actionText` plugin cards. Ability to add select text based on plugin status.
* Added `isConfigurable` prop to allow 1 click configuration.
* Added two more callbacks `deactivate` & `configure`.
* `isTogglable` functionality to allow toggle (deactivate/activate) behavior for plugin cards.

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets - `git checkout origin/feat/ia-settings-syndication && npm run build`
2. Navigate to _/wp-admin/admin.php?page=newspack-settings#/syndication_
3. Observe new tab "Syndication" loads correctly and it has 3 components; Rss, Apple News and Distributor (Appearance will vary depending on what plugin you have installed and activated.):

![Screenshot 2024-07-15 at 09 19 45](https://github.com/user-attachments/assets/0014f32b-982b-4d93-9234-f1dee2dadcfc)

4. Test:
   * Toggling on/off individual cards. Page will refresh after actions are completed (necessary for plugin related menu items to become visible/invisible).
   * Delete all related Apple News storage:
```sql
SELECT * FROM `wp_options` WHERE `option_name` LIKE '%apple_news%'
```
   * Refresh page and activate Apple News
![Screenshot 2024-07-15 at 09 33 51](https://github.com/user-attachments/assets/65d425ef-5754-4d85-991f-7bfd629c4837)
   * Run `wp option delete newspack_has_configured_plugin_publish-to-apple-news` in the terminal and refresh browser again. You should see an option to configure i.e. "Configure Apple News".
![Screenshot 2024-07-15 at 09 35 14](https://github.com/user-attachments/assets/cd0bc0e1-3d09-4527-89bd-0753f3b08c71)
   * Go ahead and click "Configure Apple News". 
   * After clicking - running `wp option get newspack_has_configured_plugin_publish-to-apple-news` should yield `1`, this is confirmation that Apple News has been configured.
![Screenshot 2024-07-15 at 09 40 02](https://github.com/user-attachments/assets/6fb45e02-c895-42fc-b5fd-5e8883177ccb)


### Other information:
